### DISCUSSION TITLE
Does the "[fuzz]" TOML object apply to all profiles by default?

### DISCUSSION THREAD

**Discussion by PaulRBerg** [foundry-rs/foundry#3571] [Help] [answered]
I noticed that the `fuzz_runs` property has been refactored to `fuzz.runs` in the latest version of Forge. So instead of this:

```toml
[profile.ci]
fuzz_runs = 1000
```

We have to write this now:

```toml
[profile.ci]
[fuzz]
runs = 1000
```

The question is - does this now apply to all profiles, or only to the `ci` profile, because the  `[fuzz]` object is defined beneath it?

I'm sorry, I imagine that this is more of a TOML parsing question than a Foundry question, though because this is a breaking change it would be worth it be clarified 🙏

My guess is that if I want to apply the `fuzz` settings only to the `ci` profile, I would have to define it as `profile.ci.fuzz`. Is that correct?

**Comment by mds1** [comment, marked as answer]
`[fuzz]` is an alias for `[profile.default.fuzz]`. The default profile config is applied to other profiles as a fallback, i.e. if they don't have their own config. So:
- You can change the fuzz config for your `ci` profile with a `[profile.ci.fuzz]` section.
- If you didn't have a `[profile.ci.fuzz]` section, the CI profile would use whatever is specified under `[fuzz]` (or equivalently, `[profile.default.fuzz]`)

[Here](https://github.com/ScopeLift/foundry-template/blob/main/foundry.toml) is an example of a foundry config that uses the default fuzz/invariant settings for the default, but changes the parameters for `lite` and `ci` profiles

**Reply by PaulRBerg** [reply]
Thanks a lot for your answer.

Just to double-check my understanding of TOML - is there no other way to selectively declare the `fuzz` section for different profiles besides the `[profile.NAME.fuzz]` way? I find this approach a bit verbose, because I have to specify the full path of the object.

**Reply by mds1** [reply]
In theory foundry could probably supporting excluding the `profile.` prat and making it implicit, but otherwise you do have to use the `[profile.NAME.SECTION]` approach. Currently there's also `profile.NAME.invariant` for invariant test config (not well documented, more info soon ™️), and there may also be a `profile.name.solidity` section as mentioned in https://github.com/foundry-rs/foundry/issues/3062

I also recommend reading https://toml.io/en/v1.0.0, it does a great job of explaining the TOML spec and alternative syntaxes

**Reply by PaulRBerg** [reply]
Thanks for the tip - I skimmed through it and I think I have what I was looking for. We can also use an inline table besides the `profile.NAME.fuzz` approach:

```toml
[profile.NAME]
# Other key/value pairs
fuzz = { runs = 256 }
```

**Reply by mds1** [reply]
Yep that definitely works too! 🙌

I think I agree inline table syntax makes things easier to read, the flat structure of TOML can be tricky to read/skim sometimes

**Reply by mds1** [reply]
While we're talking about TOML formatting, I also like using [taplo](https://taplo.tamasfe.dev/) for formatting (there's a VSCode extension + CLI tool) with `indent_entries` set to true, as I find the added indentation easier to read

**Reply by PaulRBerg** [reply]
Thanks again for the tips, ser. Very useful.
